cache/cache.go

```diff
@@ -0,0 +1,43 @@
+package cache
+
+import (
+	"main/repo"
+	"sync"
+)
+
+type CacheInterface interface {
+	Set(value *repo.Record)
+	Get(key string) *repo.Record
+	Delete(key string)
+}
+
+type InMemoCache struct {
+	cache map[string]*repo.Record
+	mu    sync.Mutex
+}
+
+func NewInMemoCache() *InMemoCache {
+	return &InMemoCache{cache: make(map[string]*repo.Record)}
+}
+
+func (c *InMemoCache) Set(value *repo.Record) {
+	c.mu.Lock()
+	c.cache[value.ID] = value
+	c.mu.Unlock()
```

Comment:
unlock via `defer` because `value` can be `nil` or check if it is not `nil` before the lock

Reply:
Anyway, cache must accept `string` parameter as a `key` and `*repo.Record` as a value. As it's defined
```
	cache map[string]*repo.Record
```